sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs

```diff
@@ -708,6 +760,50 @@ public static void SendEvents_TestHub(int numEvents, string input, [EventHub(Tes
             }
         }
 
+        public class EventHubTestMultipleDispatchMinBatchSizeJobs
+        {
+            private static int s_eventCount;
+            private static int s_processedEventCount;
+            public static async Task SendEvents_TestHub(int numEvents, string input, [EventHub(TestHubName, Connection = TestHubName)] EventHubProducerClient client)
+            {
+                // Send all of the events to the same partition so the test is deterministic
+                s_eventCount = numEvents;
+                var options = new SendEventOptions()
+                {
+                    PartitionKey = "key1"
+                };
+
+                // send one event at a time with a short time gap in between
+                for (int i = 0; i < numEvents; i++)
+                {
+                    var evt = new EventData(Encoding.UTF8.GetBytes(input));
+                    await client.SendAsync(new[] { evt }, options).ConfigureAwait(false);
+                    await Task.Delay(1000);
```

Comment:
Why do we have the delay?

Reply:
The delay is to space out the events a little to make sure they get dispatched to the `EventHubListener.PartitionProcessor` in separate calls, if they're sent to close together the host just puts them all in one array and only calls `EventHubListener.PartitionProcessor.ProcessEventsAsync` once